Using Complex in a hashset?

Hi All,

I am trying to use complex numbers (rustc v1.13.0, num v0.1.36, num-complex v0.1.35) in a HashSet, for example:

```rust
extern crate num;

use std::collections::HashSet;
use num::complex::*;

fn main() {
    let mut set = HashSet::new();
    let z = Complex::new(1, 1);
    set.insert(z);
}
```

and I get the following error;

```
error[E0277]: the trait bound `num::Complex<{integer}>: std::cmp::Eq` is not satisfied
 --> src/main.rs:8:19
  |
8 |     let mut set = HashSet::new();
  |                   ^^^^^^^^^^^^ trait `num::Complex<{integer}>: std::cmp::Eq` not satisfied
  |
  = note: required by `<std::collections::HashSet<T>>::new`

error[E0277]: the trait bound `num::Complex<{integer}>: std::cmp::Eq` is not satisfied
  --> src/main.rs:10:9
   |
10 |     set.insert(z);
   |         ^^^^^^ trait `num::Complex<{integer}>: std::cmp::Eq` not satisfied
```

I'm confused because in the documentation is says PartialEq is implemented for complex numbers, and doesn't that mean Eq is also implemented?  Any help here would be greatly appreciated!  Thanks.

--- a/complex/src/lib.rs
+++ b/complex/src/lib.rs
@@ -33,7 +33,7 @@ use traits::{Zero, One, Num, Float};
 // probably doesn't map to C's _Complex correctly.
 
 /// A complex number in Cartesian form.
-#[derive(PartialEq, Copy, Clone, Hash, Debug, Default)]
+#[derive(PartialEq, Eq, Copy, Clone, Hash, Debug, Default)]
 #[cfg_attr(feature = "rustc-serialize", derive(RustcEncodable, RustcDecodable))]
 pub struct Complex<T> {
     /// Real portion of the complex number
@@ -1218,6 +1218,20 @@ mod test {
         assert!(::hash(&a) != ::hash(&b));
         assert!(::hash(&b) != ::hash(&c));
         assert!(::hash(&c) != ::hash(&a));
+    }
+
+    #[test]
+    fn test_hashset() {
+        use std::collections::HashSet;
+        let a = Complex::new(0i32, 0i32);
+        let b = Complex::new(1i32, 0i32);
+        let c = Complex::new(0i32, 1i32);
+
+        let set: HashSet<_> = [a, b, c].iter().cloned().collect();
+        assert!(set.contains(&a));
+        assert!(set.contains(&b));
+        assert!(set.contains(&c));
+        assert!(!set.contains(&(a + b + c)));
     }
 
     #[test]

